File: src/main/java/com/natia/secretmod/hooks/RenderHook.java
package com.natia.secretmod.hooks;

import com.natia.secretmod.features.DamagePerSecond;
import com.natia.secretmod.features.TimersHook;
import com.natia.secretmod.features.altC.QuickTab;
import com.natia.secretmod.features.fishing.WormCounter;
import com.natia.secretmod.features.fishing.WormTimer;
import com.natia.secretmod.features.slayers.RNGesusBar;
import com.natia.secretmod.features.slayers.VoidGloom;
import net.minecraftforge.fml.common.eventhandler.EventPriority;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import net.minecraftforge.fml.common.gameevent.TickEvent;

public class RenderHook {

    @SubscribeEvent(priority = EventPriority.HIGHEST)
    public void render(TickEvent.RenderTickEvent event) {
        VoidGloom.getInstance().render();
        TimersHook.getInstance().render();
        WormCounter.getInstance().render();
        WormTimer.getInstance().render();
        RNGesusBar.getInstance().render();
        DamagePerSecond.getInstance().render();
    }

}

File: src/main/kotlin/natia/skytweaks/hooks/RenderHook.kt
package natia.skytweaks.hooks

import natia.skytweaks.features.DamagePerSecond
import natia.skytweaks.features.SummonsAlert
import natia.skytweaks.features.TimersHook
import natia.skytweaks.features.dungeons.BatPointer
import natia.skytweaks.features.fishing.WormCounter
import natia.skytweaks.features.fishing.WormTimer
import natia.skytweaks.features.slayers.RNGesusBar
import natia.skytweaks.features.slayers.VoidGloom
import net.minecraft.client.Minecraft
import net.minecraftforge.client.GuiIngameForge
import net.minecraftforge.client.event.RenderGameOverlayEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

class RenderHook {

    @SubscribeEvent
    fun render(event: RenderGameOverlayEvent.Post) {
        /* prevent labymod from fucking everything up */
        if (Minecraft.getMinecraft().ingameGUI is GuiIngameForge) {
            if (event.type == RenderGameOverlayEvent.ElementType.EXPERIENCE || event.type == RenderGameOverlayEvent.ElementType.JUMPBAR) {
                VoidGloom.instance.render()
                TimersHook.instance.render()
                WormCounter.instance.render()
                WormTimer.instance.render()
                RNGesusBar.instance.render()
                DamagePerSecond.instance.render()
                BatPointer.instance.render()
//                SummonsAlert.instance.render()
            }
        }
    }

}
